### ISSUE TITLE
[BUG] tags passed to log_model are not propagated to the registered ModelVersion when registered_model_name is specified

### ISSUE THREAD

**Issue by vrcadore** [mlflow/mlflow#24101]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

- Client: 3.14.0
- Tracking server: 3.14.0

### System information

- **OS Platform and Distribution**: Ubuntu 24.04
- **Python version**: 3.11

### Describe the problem

When calling `mlflow.<flavor>.log_model(..., tags={"key": "value"}, registered_model_name="my-model")`, the `tags` argument is silently dropped and never attached to the resulting `ModelVersion`. The tags end up only on the `LoggedModel` entity, which is invisible when inspecting the model registry.

### Root cause

In `mlflow.models.model.Model.log` (`mlflow/models/model.py`), after the `LoggedModel` is created with the user-supplied tags, `_register_model` is called **without** forwarding `tags`:

```python
# model.py – Model.log()
if registered_model_name is not None:
    registered_model = mlflow.tracking._model_registry.fluent._register_model(
        f"models:/{model.model_id}",
        registered_model_name,
        await_registration_for=await_registration_for,
        local_model_path=local_path,
        # tags is NOT forwarded here ← bug
    )
```

`_register_model` already accepts a `tags` keyword argument and forwards it correctly to `client._create_model_version`. The fix is a one-liner: add `tags=tags` to the call above.

### Expected behaviour

Tags passed to `log_model` should be attached to both the `LoggedModel` entity and the registered `ModelVersion` (when `registered_model_name` is provided), consistent with what a caller would intuitively expect.

### Proposed fix

```python
# model.py – Model.log()
if registered_model_name is not None:
    registered_model = mlflow.tracking._model_registry.fluent._register_model(
        f"models:/{model.model_id}",
        registered_model_name,
        await_registration_for=await_registration_for,
        local_model_path=local_path,
        tags=tags,  # ← forward tags to ModelVersion
    )
````

### Code to reproduce issue

```python
import mlflow
import mlflow.sklearn
from sklearn.linear_model import LinearRegression

mlflow.set_tracking_uri("http://localhost:5000")

model = LinearRegression().fit([[1], [2]], [1, 2])

with mlflow.start_run():
    info = mlflow.sklearn.log_model(
        model,
        name="my-model",
        registered_model_name="my-model",
        tags={"stage": "training", "framework": "sklearn"},
    )

client = mlflow.MlflowClient()

# Tags ARE on the LoggedModel
print(client.get_logged_model(info.model_id).tags)
# → {'stage': 'training', 'framework': 'sklearn'}

# Tags are NOT on the ModelVersion
mv = client.get_model_version("my-model", info.registered_model_version)
print(mv.tags)
# → {}  ← expected {'stage': 'training', 'framework': 'sklearn'}

```

### Stack trace

No errors on stack trace.

### What component(s) does this bug affect?

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

**Comment by Ankita7033** [comment]
I've identified the root cause and have a fix ready. In Model.log() (mlflow/models/model.py), _register_model(...) is called without forwarding tags, so tags passed to log_model(..., registered_model_name=...) never reach the ModelVersion. I've made the one-line fix (tags=tags) and added a regression test that confirms the ModelVersion now carries the tags. Could a maintainer triage this and add the ready label so I can open the PR? Happy to proceed as soon as it's labeled.

**Comment by vrcadore** [comment]
I am happy @Ankita7033  if you open a PR and submit a fix to it.

**Comment by Ankita7033** [comment]
Thanks @vrcadore! I have the fix and a regression test ready to go. Per the contributing policy, could a maintainer triage this and add the ready label so I can open the PR without it being auto-closed? Happy to submit immediately once it's labeled.

**Comment by PattaraS** [comment]
Thanks @vrcadore for the clear root-cause, and @Ankita7033 for picking it up. Confirmed the bug: `_register_model` accepts a tags kwarg and forwards it to the `ModelVersion`, but `Model.log` calls it without passing tags, so they only ever land on the `LoggedModel`.

One thing to watch - there are two call sites, not one:
  - `mlflow/models/model.py:1367` (`Model.log`, `models:/ URI`) — the one in the issue; tags is in scope, so tags=tags works directly.
  - `mlflow/models/model.py:1041` (`_log_v2`, `runs:/ URI`) — same omission. Worth confirming tags is threaded through here too so both paths are covered.

Marking ready. @Ankita7033 it's yours - please include the regression test you mentioned, and cover both call sites if the _log_v2 path applies.
